vyper/exceptions.py

```diff
@@ -70,3 +70,11 @@ class InvalidTypeException(ParserException):
 
 class TypeMismatchException(ParserException):
     pass
+
+
+class FunctionDeclarationException(ParserException):
+    pass
+
+
+class EventDeclarationException(ParserException):
+    pass
```

Comment:
So was about to merge, but then occurred to me that if we introduce these two exceptions we have to use them more than just in the added unicode checks.
So please take the other VariableDeclarationException or similar exceptions that occur on on function and event declaration and use the above two exceptions. This way the use of the exceptions are consistent.